src/Components/Notifications/PartnerOfferArtwork.tsx

```diff
@@ -0,0 +1,144 @@
+import { FC } from "react"
+import { graphql, useFragment } from "react-relay"
+import { PartnerOfferArtwork_artwork$key } from "__generated__/PartnerOfferArtwork_artwork.graphql"
+import { resized } from "Utils/resized"
+import { ManageArtworkForSavesProvider } from "Components/Artwork/ManageArtworkForSaves"
+import { RouterLink } from "System/Router/RouterLink"
+import { Box, Button, Flex, Image, Text } from "@artsy/palette"
+import Metadata from "Components/Artwork/Metadata"
+import { ContextModule } from "@artsy/cohesion"
+import { useTimer } from "Utils/Hooks/useTimer"
+import { CARD_MAX_WIDTH } from "Components/Notifications/constants"
+
+interface PartnerOfferArtworkProps {
+  artwork: PartnerOfferArtwork_artwork$key
+  targetHref: string
+  endAt?: string | null
+  available?: boolean | null
+  priceListedMessage?: string | null
+  priceWithDiscountMessage?: string | null
+}
+
+export const PartnerOfferArtwork: FC<PartnerOfferArtworkProps> = ({
+  artwork: artworkProp,
+  targetHref,
+  priceListedMessage,
+  priceWithDiscountMessage,
+  endAt = "",
+  available = false,
+}) => {
+  const { hasEnded } = useTimer(endAt || "")
+  const fullyAvailable = !!(
+    available &&
+    !hasEnded &&
+    priceWithDiscountMessage &&
+    priceListedMessage
+  )
+
+  const artwork = useFragment(partnerOfferArtworkFragment, artworkProp)
+  const image = resized(artwork?.image?.src ?? "", { width: CARD_MAX_WIDTH })
+  const label =
+    (artwork.title ?? "Artwork") +
+    (artwork.artistNames ? ` by ${artwork.artistNames}` : "")
+
+  let buttonText = "Continue To Purchase"
+  if (hasEnded) buttonText = "View Work"
+  if (!available) buttonText = "Create Alert"
+
+  let href = targetHref
+  if (!available) {
+    href = `${artwork.href}${
+      artwork.href?.includes("?") ? "&" : "?"
+    }unavailable=true`
+  } else if (hasEnded) {
+    href = `${artwork.href}${
+      artwork.href?.includes("?") ? "&" : "?"
+    }expired_offer=true`
+  }
```

Comment:
It might be good to use JavaScript's [URL interface](https://developer.mozilla.org/en-US/docs/Web/API/URL) here:

```suggestion
  let url = new URL(targetHref);

  if (!available) {
    url = new URL(artwork.href)
    url.searchParams.append("unavailable", "true");
  } else if (hasEnded) {
    url = new URL(artwork.href)
    url.searchParams.append("expired_offer", "true");
  }

  const href = url.href;
```